inc/user.class.php

```diff
@@ -3404,173 +3402,243 @@ static function getSqlSearchResult ($count = true, $right = "all", $entity_restr
-                                  getEntitiesRestrictRequest("AND", "glpi_profiles_users", '',
-                                                             $entity_restrict, 1).") ";
+                     $WHERE[] = [
+                        'OR' => [
+                           'AND' => [
+                              'glpi_profilerights.name'     => $r,
+                              'glpi_profilerights.rights'   => [
+                                 '&',
+                                 (READ | CREATE | UPDATE | DELETE | PURGE)
+                              ]
+                           ] + getEntitiesRestrictCriteria('glpi_profiles_users', '', $entity_restrict, 1)
+                        ]
+                     ];
                }
                if (in_array($r, Profile::$helpdesk_rights)) {
                   $forcecentral = false;
                }
             }
 
-            $where = '('.implode(' OR ', $where);
-
             if ($forcecentral) {
-               $where .= " AND `glpi_profiles`.`interface` = 'central' ";
+               $WHERE['glpi_profiles.interface'] = 'central';
             }
-            $where .= ')';
       }
 
       if (!$inactive_deleted) {
-         $where .= " AND `glpi_users`.`is_deleted` = 0
-                     AND `glpi_users`.`is_active` = 1
-                     AND (`glpi_users`.`begin_date` IS NULL
-                          OR `glpi_users`.`begin_date` < NOW())
-                     AND (`glpi_users`.`end_date` IS NULL
-                          OR `glpi_users`.`end_date` > NOW())";
+         $WHERE = array_merge(
+            $WHERE, [
+               'glpi_users.is_deleted' => 0,
+               'glpi_users.is_active'  => 1,
+               [
+                  'OR' => [
+                     ['glpi_users.begin_date' => null],
+                     ['glpi_users.begin_date' => ['<', new QueryExpression('NOW()')]]
+                  ]
+               ],
+               [
+                  'OR' => [
+                     ['glpi_users.end_date' => null],
+                     ['glpi_users.end_date' => ['>', new QueryExpression('NOW()')]]
+                  ]
+               ]
+
+            ]
+         );
       }
 
       if ((is_numeric($value) && $value)
           || count($used)) {
 
-         $where .= " AND `glpi_users`.`id` NOT IN (";
-         if (is_numeric($value)) {
-            $first  = false;
-            $where .= $value;
-         } else {
-            $first = true;
-         }
-         foreach ($used as $val) {
-            if ($first) {
-               $first = false;
-            } else {
-               $where .= ",";
-            }
-            $where .= $val;
-         }
-         $where .= ")";
+         $WHERE[] = [
+            'NOT' => [
+               'glpi_users.id' => $used
+            ]
+         ];
       }
 
+      $criteria = [
+         'SELECT'    => 'glpi_users.*',
```

Comment:
The distinct on `*` is important when using left joins (see https://www.db-fiddle.com/f/7Q39KHLyxMbh7KsYsypar2/1)

Reply:
Yes, I saw that working on another part... :/